benchmark/fs/readfile-partitioned.js

```diff
@@ -0,0 +1,84 @@
+// Call fs.readFile over and over again really fast.
+// Then see how many times it got called.
+// Also report how many times we were able to run a small concurrent
+// operation on the threadpool.
+'use strict';
+
+const path = require('path');
+const common = require('../common.js');
+const filename = path.resolve(__dirname,
+                              `.removeme-benchmark-garbage-${process.pid}`);
+const fs = require('fs');
+const zlib = require('zlib');
+const assert = require('assert');
+
+const bench = common.createBenchmark(main, {
+  dur: [5],
+  len: [1024, 16 * 1024 * 1024],
+  concurrent: [1, 10]
+});
+
+function main(conf) {
+  const len = +conf.len;
+  try { fs.unlinkSync(filename); } catch (e) {}
+  var data = Buffer.alloc(len, 'x');
+  fs.writeFileSync(filename, data);
+  data = null;
+
+  var zipData = Buffer.alloc(1024, 'a');
+
+  var reads = 0;
+  var zips = 0;
+  var benchEnded = false;
+  bench.start();
+  setTimeout(function() {
+    const totalOps = reads + zips;
+    console.log('bench ended: ' +
+      `reads ${reads}, zips ${zips}, total ops ${totalOps}`);
+    benchEnded = true;
+    bench.end(totalOps);
+    try { fs.unlinkSync(filename); } catch (e) {}
+  }, +conf.dur * 1000);
+
+  function read() {
+    fs.readFile(filename, afterRead);
+  }
+
+  function afterRead(er, data) {
+    if (er) {
+      if (er.code === 'ENOENT') {
+        // Race with timer.
+        assert.assert(benchEnded);
```

Comment:
This failed in CI tests because `assert.assert()` is not a function. It should be `assert.ok(benchEnded)`.

Reply:
Ahha I'd been wondering why the CI was failing. Will fix, thank you.

Reply:
Addressed in d359b35.